syft/frameworks/torch/fl/dataset.py

```diff
@@ -253,6 +253,11 @@ def workers(self):
 
         return list(self.datasets.keys())
 
+    def get_dataset(self, worker):
+        dataset = self[worker].get()
+        del self.datasets[worker]
```

Comment:
Do we really want that? What is the usecase for `get_dataset()`?

Reply:
because if the user used federated_dataset.datasets[worker].get() that would leave it with a pointer that points to nothing. that will require the pointer to be deleted from the datasets dict if we want the federated_dataset object to still work. the user won't be able to do it if he doesn't have a deep understanding of the implementation because it will cause errors in other parts of his code like dataset_loaders for example. this is a way to keep the object always clean and ready to use